### DISCUSSION TITLE
Link to a pdf doc.

### DISCUSSION THREAD

**Discussion by brianshano** [vercel/next.js#25981] [Help] [answered]
I want to put a pdf file in my public/ folder and create a simple `<a href>` link for the user to download it. 
It's proving more complicated than I expected!

`You may need an appropriate loader to handle this file type`

Do I really need a loader for it?   Or is there a simple way to link to a pdf?

**Comment by rootedsoftware** [comment, marked as answer]
You can just put the file in the /public directory and then you use something like this

If your file is stored in public/files/ use the below href syntax

```
  <a
    href="/files/file_name.pdf"
    alt="alt text"
    target="_blank"
    rel="noopener noreferrer"
  >Download FIle</a>
```

Don't include "public" in the href path.

**Reply by RachaelSMathew** [reply]
I tried this and when I click on the link I'm getting sent to a 404 link that states the pdf doesn't exist.

**Reply by hschhhwwwo0o** [reply]
> I tried this and when I click on the link I'm getting sent to a 404 link that states the pdf doesn't exist.

Place in the root of the `public` folder. It helped me.

**Reply by webdiego** [reply]
Thanks

**Reply by rayliao** [reply]
How to solve with i18n? It won't work with like `url/fr/file_name.pdf`, 404

**Reply by JoelBonetR** [reply]
The best approach I could think of is to set it like: `host/file_name_fr.pdf`, then you use the context language to concatenate: `file_name_[lang}.pdf` when generating the link in the UI.

**Reply by rayliao** [reply]
I solved it last night, just set locale to false，like: `<Link href="/file_name.pdf" locale={false}>...</Link>`, Thank you for answer anyway :)

**Reply by EugeneGohh** [reply]
thanks!

**Reply by ArtimusMaximus** [reply]
In case this helps someone else, my issue was:

import **tableDataPDF** from 'public/images/table_data.pdf'
`<Link  href={tableDataPDF}  target="_blank"  rel="noopener noreferrer" locale={false} download>Download Here</Link>`
changed to:
`<Link  href={'public/images/table_data.pdf'}  target="_blank"  rel="noopener noreferrer" locale={false} download>Download Here</Link>`

Putting the import there instead of the string literal was causing issues.

**Reply by WebDevZakaria** [reply]
thanks it works!

**Comment by tariksahni** [comment]
You have to add `file-loader` to webpack. Teach webpack how you can bundle pdf files.
Edit: `next.config.js`


```
module.exports = {
  reactStrictMode: true,
  webpack: (config) => {
    config.module.rules.push({
      test: /\.pdf$/,
      use: {
        loader: 'file-loader',
        options: {
          name: '[path][name].[ext]',
        },
      },
    });
    return config;
  }
}
```

**Comment by SiddheshDabholkar** [comment]
you an do it by using **download** attribute in `a` tag ie.,


If your file is stored in **public** folder than use the following href syntax

```
 <a download href="/resume.pdf">
    Resume
 </a>
```

**Comment by tyler-morales** [comment]
You have to make sure you use a relative path. Instead of `/public/...`, use `/files/file.pdf`

**Reply by Shwrndasink** [reply]
This helped thanks!

**Reply by gitanasageaura** [reply]
thank you

**Reply by Kashyap1ankit** [reply]
This thing only helped me... try it out

**Comment by fadhilradh** [comment]
what if the file is in third party bucket (i.e. AWS, Google Storage) ? Anyone have solution ?

**Reply by Artem66** [reply]
https://github.com/vercel/next.js/discussions/25981#discussioncomment-8159401

**Comment by BulochkaBU** [comment]
Today I try to make link like this
` <a download href="/resume.pdf"> Resume </a>`
In downloads i see file resume.pdf. but it doesn't open because actually file extension resume.pdf.html, i don't know why and how to fix it
I use Next.js App Router

**Reply by BulochkaBU** [reply]
My problem was happened because I used i18n, and my URL like 'localhost:3000/en/resume.pdf'. And I created new folder 'en' in folder 'public', and then it works

**Comment by VittorioMorellini** [comment]
I have this problem on Next14, I am not able to have a download for a file pdf as a static asset

**Reply by khatiwadaaj** [reply]
If you haven't already fixed it you can try the following which worked for me in Next 13.5.4.

I ran into same issue with the following code, it was working in development but not after I deployed it to Github pages.
<a download href=**'/CV.pdf'**>
   Download CV
</a>

CV.pdf is located in root of public, /public.

I changed it to 
<a download href=**'./CV.pdf'**>
    Download CV
 </a>

**Comment by mushfiqur-rahman** [comment]
I tried all the method mention but still getting error like

`./public/filename.pdf
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)`

and my code is
`<div className="grid md:grid-cols-2 gap-2">
            
              <a
                href="../public/Mushfiqur_Rahman_Resume.pdf"
                alt="alt text"
                target="_blank"
                rel="noopener noreferrer"
              >
                Download FIle
              </a>
            </div>`
            
**Or,**


            `<div className="grid md:grid-cols-2 gap-2">
              {/* Latex resume */}
              <Link
                className="py-4"
                href="../public/Mushfiqur_Rahman_Resume.pdf"
                locale={false}
                rel="noopener noreferrer"
                target="_blank"
                aria-label="Downlod Resume"
              >
                <button className="p-5 font-bold hover:text-white">
                  Downlod Resume
                </button>
              </Link>
            </div>`

**Reply by mushfiqur-rahman** [reply]
Problem is solved. My Code now

`<Link
                className="py-4"
                href="/Mushfiqur_Rahman_Resume.pdf"
                locale={false}
                rel="noopener noreferrer"
                target="_blank"
                aria-label="Downlod Resume"
              >
                <button className="p-5 font-bold hover:text-white">
                  Downlod Resume
                </button>
              </Link>`

**Comment by Artem66** [comment]
That helps me in the Next JS app


I created the file download-image.js

```  
// pages/api/download-image.js
import axios from 'axios';
export default async function handler(req, res) {
  try {
    const { imageUrl } = req.query;

    if (!imageUrl) {
      return res.status(400).send('Missing imageUrl parameter');
    }

    const imageResponse = await axios.get(imageUrl, { responseType: 'arraybuffer' });

    // Set appropriate headers for image response
    res.setHeader('Content-Type', 'image/jpeg');
    res.setHeader('Content-Disposition', `attachment; filename="${imageUrl.split('/').pop()}"`);

    // Send the image data to the client
    res.send(imageResponse.data);
  } catch (error) {
    console.error(error);
    res.status(500).send('Internal Server Error');
  }
}  
```

and use this in component smth like this 

```
<Link
   href={`/api/download-image?imageUrl=${image.mediaItemUrl}`}
   download={image.mediaItemUrl.split('/').pop()} 
 > 
  Download
</Link>
```

**Comment by YounesSensie** [comment]
Hello, How do I display a PDF but prevent it from downloading? when i was  using nextjs and react-pdf but i faild to prevent medium user to download it because his able to go to the google tools inn the goole chrome and find the link of pdf inside network of goole dev tools

**Comment by sehrish30** [comment]
` <Link
                  href="/sehrish.pdf"
                  download="Sehrish.pdf"
                  locale={false}
                  rel="noopener noreferrer"
                  target="_blank"
                  aria-label="Downlod Resume"
                  className={styles["link--secondary"]}
                >
                  My Resume
                </Link>`
                
    in next.js 14

**Reply by aaron-pn** [reply]
Thanks :D

**Comment by monolithed** [comment]
It's a bit odd that file downloading only works through the Link tag. I tried using the standard 'a' tag, but Vercel always returns a 404 error.
